### PR TITLE
Fix typos in comments

### PR DESCRIPTION
**Pull request by spaette**
This fixes primarily double slash XML C# comments.

Pay special attention to whether the changes to the following files are acceptable.

src/System.Management.Automation/engine/parser/Compiler.cs
src/System.Management.Automation/engine/runtime/Binding/Binders.cs
src/System.Management.Automation/security/CatalogHelper.cs
src/System.Management.Automation/engine/ComInterop/IDispatchComObject.cs
src/Microsoft.Management.UI.Internal/HelpWindow/ParagraphBuilder.cs

### CODE REVIEW

**Comment by spaette** [comment]
Per request I can scan files with extensions other than .cs and .resx for typographical errors.

**Comment by spaette** [comment]
@iSazonov 

Is there a basis to rerun the 2 failing checks?

File: src/Microsoft.Management.UI.Internal/HelpWindow/ParagraphSearcher.cs [hunk 1]
```diff
@@ -43,8 +43,8 @@ internal ParagraphSearcher()
         /// <returns>The next highlight starting at the <paramref name="caretPosition"/>.</returns>
         internal Run MoveAndHighlightNextNextMatch(bool forward, TextPointer caretPosition)
         {
-            Debug.Assert(caretPosition != null, "a caret position is allways valid");
-            Debug.Assert(caretPosition.Parent != null && caretPosition.Parent is Run, "a caret PArent is allways a valid Run");
+            Debug.Assert(caretPosition != null, "a caret position is always valid");
+            Debug.Assert(caretPosition.Parent != null && caretPosition.Parent is Run, "a caret parent is always a valid Run");
```
Comment (iSazonov) [inline src/Microsoft.Management.UI.Internal/HelpWindow/ParagraphSearcher.cs:47]: I guess it is reference to property name.
```suggestion
            Debug.Assert(caretPosition.Parent != null && caretPosition.Parent is Run, "a caret Parent is always a valid Run");
```

**Review by iSazonov — APPROVED** [review]
LGTM with one comment.

**Review by PaulHigin — APPROVED** [review]

**Comment by spaette** [comment]
will do a soft reset

**Comment by spaette** [comment]
my soft reset shouldn't result in the check fails

**Comment by iSazonov** [comment]
Reopen to restart CIs.

**Comment by iSazonov** [comment]
@spaette Thanks for your contribution!

**Comment by spaette** [comment]
to be precise these typo corrected strings weren't found in comments

AmbiguosAuthentication
IsHomogenousArray
sucessfulDecompression

and there were some typo fixes of strings which had appeared enclosed in double quotes

also hopefully some textual changes for the purpose of readability

**Pull request merged**